### PR TITLE
[buffer-crc32] Improve tests and docs

### PR DESCRIPTION
**Pull request by BendingBender**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/brianloveswords/buffer-crc32
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/buffer-crc32/index.d.ts [hunk 1]
```diff
@@ -7,9 +7,51 @@
...
+     * Convenience method that returns a signed int instead of a `Buffer`.
+     *
+     * @example
+     * import crc32 = require('buffer-crc32');
+     *
+     * // works with buffers
+     * const buf = Buffer.from([0x00, 0x73, 0x75, 0x70, 0x20, 0x62, 0x72, 0x6f, 0x00]);
+     * crc32.signed(buf); // -> -1805997238
+     */
     function signed(buffer: string | Buffer, partialCrc?: Buffer | number): number;
```
Comment (NoodleOfDeath) [inline types/buffer-crc32/index.d.ts:44]: Doesn't this need to be exported? 🤔

File: types/buffer-crc32/index.d.ts [hunk 2]
```diff
@@ -7,9 +7,51 @@
...
+     * Convenience method that returns an unsigned int instead of a `Buffer`.
+     *
+     * @example
+     * import crc32 = require('buffer-crc32');
+     *
+     * // works with buffers
+     * const buf = Buffer.from([0x00, 0x73, 0x75, 0x70, 0x20, 0x62, 0x72, 0x6f, 0x00]);
+     * crc32.unsigned(buf); // -> 2488970058
+     */
     function unsigned(buffer: string | Buffer, partialCrc?: Buffer | number): number;
```
Comment (NoodleOfDeath) [inline types/buffer-crc32/index.d.ts:56]: This too?

File: types/buffer-crc32/buffer-crc32-tests.ts [hunk 3]
```diff
@@ -1,19 +1,23 @@
 import crc32 = require('buffer-crc32');
```
Comment (NoodleOfDeath) [inline types/buffer-crc32/buffer-crc32-tests.ts:1]:
```suggestion
import crc32 from 'buffer-crc32';
```

**Review by NoodleOfDeath — COMMENTED** [review]
Just added some comments. Found this thread while trying to research whether or not namespaces should ever be used

File: types/buffer-crc32/index.d.ts (hunk 1 above)
Comment (BendingBender) [reply types/buffer-crc32/index.d.ts:44]: nope, contents of namespaces are exported automatically

File: types/buffer-crc32/index.d.ts (hunk 2 above)
Comment (BendingBender) [reply types/buffer-crc32/index.d.ts:56]: nope, contents of namespaces are exported automatically

File: types/buffer-crc32/buffer-crc32-tests.ts (hunk 3 above)
Comment (BendingBender) [reply types/buffer-crc32/buffer-crc32-tests.ts:1]: No, we have a commonjs export here, so the only way to import that'll work correctly everywhere (independent of tsconfig settings) is a commonjs import.

**Review by NoodleOfDeath — APPROVED** [review]

**Comment by BendingBender** [comment]
Ready to merge

**Pull request merged**